Fix issue #268 : wrong shirt colors in the character selection

Racater opened the pull request:
This pull request fixes the issue #268 .
Now the characters will get the correct shirt color for shirts such as the space suit from the server following character creation (dark green and medium blue).
The error was probably due to a wrong copy and paste in the code.
Has been tested locally.

averysumner commented:
Looks good to me.

Jettford commented:
Maybe this should be replaced with a CDClient query such as this implementation of it. https://github.com/UchuServer/Uchu/blob/92894ace6f0f83ff1c9725b2bde58dfa4e883481/Uchu.Char/Handlers/CharacterHandler.cs#L97-L110

DarwinAnim8or commented:
@Racater hadn't seen your latest commit before reviewing, have you tested this works?

Racater commented:
@Jettford following your comment, I rewrote and simplified the code for the shirt at character creation. This way we remove this list of hardcoded values.

Racater commented:
This has been tested to work on my local server with all edge cases I could think of (first shirt style, last shirt style, different colors, etc.)

MickVermeulen commented on dGame/UserManager.cpp:
```diff
@@ -566,114 +566,34 @@ void UserManager::LoginCharacter(const SystemAddress& sysAddr, uint32_t playerID
-
-	// This will be the final shirt ID
-	uint32_t shirtIDFinal;
+uint32_t getShirtColorId(uint32_t color) {
```
This should be CamelCase

Racater replied:
done

MickVermeulen commented on dGame/UserManager.cpp:
```diff
@@ -566,114 +566,34 @@ void UserManager::LoginCharacter(const SystemAddress& sysAddr, uint32_t playerID
+uint32_t getShirtColorId(uint32_t color) {
+	
+	// get the index of the color in shirtColorVector
+	auto colorId = std::find (shirtColorVector.begin(), shirtColorVector.end(), color);
```
`std::find (` should be `std::find(`

Racater replied:
fixed

MickVermeulen reviewed (changes requested):
Have some code style comments.

MickVermeulen reviewed (approved):
Yeah LGTM, @DarwinAnim8or you assigned yourself for review, do you still plan on doing that or was it a misclick (given that you already approved before :P)
